react/features/app/middleware.js

```diff
@@ -66,6 +72,57 @@ function _connectionEstablished(store, next, action) {
     return result;
 }
 
+/**
+ * Performs an immediate reload of the application if a "split-brain" scenario
+ * is detected. The detection is based on the assumption that a connection error
+ * with "item-not-found" that happens early on in the call is a signal of
+ * split-brain.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {Dispatch} next - The redux {@code dispatch} function to dispatch the
+ * specified {@code action} to the specified {@code store}.
+ * @param {Action} action - The redux action {@code CONNECTION_FAILED}
+ * which is being dispatched in the specified {@code store}.
+ * @private
+ * @returns {Object|undefined} The new state that is the result of the reduction
+ * of the specified {@code action} or undefined if a reload is triggered.
+ */
+function _connectionFailed({ dispatch, getState }, next, action) {
+    const { error } = action;
+
+    const { timeEstablished } = getState()['features/base/connection'];
+    const { _immediateReloadThreshold } = getState()['features/base/config'];
+    const reloadThreshold = typeof _immediateReloadThreshold === 'number'
+        ? _immediateReloadThreshold : 1500;
+    const isItemNotFoundError = error
+        && error.message === 'item-not-found'
+        && error.details
+        && error.details.shard_changed;
+
+    if (isItemNotFoundError) {
+        const timeSinceConnectionEstablished
+            = timeEstablished && Date.now() - timeEstablished;
+        const isPotentialSplitBrain = !timeEstablished
+            || timeSinceConnectionEstablished <= reloadThreshold;
+
+        sendAnalytics(createConnectionEvent('failed', {
+            ...error,
```

Comment:
Analytics might not be able to handle nested objects. As such, error.details might not get logged properly. Any preference on what gets logged from the error?

Reply:
Actually even we have other events, the splitBrain field is enough to destinguish between them. So I think this is good.